src/main/java/restaurant_use_case/interactors/CreateRestaurantInteractor.java

```diff
@@ -54,7 +54,11 @@ public RestaurantResponseModel create(RestaurantRequestModel requestModel) {
         if (restaurantGateway.existsByLocation(requestModel.getLocation())) {
             // Checks if there is another restaurant at that same location, meaning they inputed the wrong location
             return presenter.prepareFailView("INVALID LOCATION");
-        } else if (requestModel.getLocation().length() == 0 |
+        } else if (!requestModel.getLocation().matches("^(?!.*[DFIOQU])[A-VXY][0-9][A-Z] ?[0-9][A-Z][0-9]$")){
```

Comment:
I should also implement this for search right? filtering by region from first letter psotal code